test/resources/DataFormats.py

```diff
@@ -79,3 +80,19 @@ def yaml_merge(base, addition):
         parsed_addition = yaml.safe_load(addition)
     _merge(combined, parsed_addition)
     return yaml.dump(combined)
+
+
+# lvmd_merge idempotently patches the lvmd config. If the patch has already
+# been applied, the base config is returned without mutation.
```

Comment:
Yeah. The datastruct is a dict of list of dicts. Yaml merge doesn't compare the list elements against each other, so it was just concatenating new thin-vol device classes rather than overwriting.  Hence the need to check the name doesn't already appear in the list.

Reply:
OK, so it's not that the patch has already been applied it's that the merge function we're using does not patch the config in the way we need for this case? The comment may just be a little misleading then.

Reply:
+1 I've updated the comment